src/core/server/saved_objects/service/lib/repository.ts

```diff
@@ -1515,26 +1515,62 @@ export class SavedObjectsRepository {
   }
 
   /**
-   * Increases a counter field by one. Creates the document if one doesn't exist for the given id.
+   * Increments all the specified counter fields by one. Creates the document
+   * if one doesn't exist for the given id.
+   *
+   * @remarks
+   * When using incrementCounter for collecting usage data, you need to ensure
+   * that usage collection happens on a best-effort basis and doesn't
+   * negatively affect your plugin or users (see the example):
+   *  - Swallow any exceptions thrown from the incrementCounter method and log
+   *    a message in development.
+   *  - Don't block your application on the incrementCounter method (e.g.
+   *    don't use `await`)
```

Comment:
I love this remark 🧡

Reply:
Isn't that a bit misplaced though? I would expect that kind of comment/guidelines to be in the usage collection plugin's README instead of being documented in a core API?

Reply:
yeah I agree, this doesn't really belong in the SavedObjectsRepository's documentation. But I everywhere where this method was called teams awaited this promise and I'm afraid they won't read the docs in the plugin's readme.

We could maybe move this to the usage data plugin's docs and add a link to the docs. Because the link will be from tsdocs -> markdown it might be best to link to the github url for that file. WDYT?

Reply:
Yea, I would go with something like

```
When using incrementCounter for collecting usage data, you need to ensure 
that usage collection happens on a best-effort basis and doesn't  negatively 
affect your plugin or users. Please refer to the [usage collection documentation](link) for more details.
```

Reply:
Done